components/sendImageForm.tsx

```diff
@@ -0,0 +1,88 @@
+'use client'
+
+import type React from 'react'
+import { useRef, useState } from 'react'
+import { ImageInput } from './imageInput'
+
+const IMAGE_ID = 'image'
+
+export const SendImageForm: React.FC = () => {
+  const fileInputRef = useRef<HTMLInputElement>(null)
+
+  // 添付画像を状態管理
+  const [imageFile, setImageFile] = useState<File | null>(null)
+  const [fileName, setFileName] = useState('')
+  const [imageSource, setImageSource] = useState('')
```

Comment:
```suggestion
  // 添付画像を状態管理
  const [images, setImages] = useState<{
    file: File | null
    name: string
    source: string
  }>({
    file: null,
    name: '',
    source: '',
  })

  // ファイルが読み込まれた際に、画像データを抽出する処理
  const generateImageSource = (files: FileList) => {
    const file = files[0]
    const fileReader = new FileReader()
    fileReader.onload = () => {
      setImages({
        ...images,
        file: file,
        name: file.name,
        source: fileReader.result as string,
      })
    }
    fileReader.readAsDataURL(file)
  }

  const handleFileChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    const files = e.target.files
    if (!files || files.length <= 0) {
      return
    }
    // 画像データを抽出する処理
    generateImageSource(files)
  }
```
Refactored

Reply:
I think your code is ok and commited it.